### PR TITLE
Populate kernel help links

### LINKED ISSUE #3177: "Help -> Python Reference" documentation should point to notebook's python version
I'm running condo-forge's jupyterlab distribution on python 3.6. When I select the python reference, it points to python 3.5.2 - great to have the quick access!

Would it be easily possible to determine the python version of the Notebook (if a python notebook or the server) and select the documentation version accordingly?

This might even happen on the npm compile time, using the default python version of the Jupiter server.

Thanks.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Adds a new semantic extension point for the `Help` menu, and populates a portion of the `Help` menu according to the kernel info links. Fixes #3177, Addresses #3103

### CODE REVIEW

File: packages/services/src/kernel/messages.ts [hunk 1]
```diff
@@ -371,7 +371,7 @@ namespace KernelMessage {
     implementation_version: string;
     language_info: ILanguageInfo;
     banner: string;
-    help_links: { [key: string]: string; };
+    help_links: { text: string; url: string }[];
```
[the remark this answers is not in the document]
Comment (blink1073) [reply packages/services/src/kernel/messages.ts]: Sweeping up them :bug: :bug:

**Comment by blink1073** [comment]
Let's not merge this until after 0.30 final.

**Comment by ian-r-rose** [comment]
One issue: the current (static) labels for the help links show `* Reference`, but the labels that come from the IPython kernel just show `*` (i.e. `Matplotlib Reference` vs `Matplotlib`). I think the former is clearer. 

However, I think it is probably better to allow the help link provider to specify their labels, rather than do any kind of string interpolating. Other kernels provide different formats for their help links that do not fit any clear string interpolation pattern. For instance, the Julia Kernel has `Julia Homepage`, `Julia Documentation`, and `Julia Packages`.

So I think that edits to the labels are better done in ipykernel rather than here.

**Comment by ian-r-rose** [comment]
Sure, no rush.

**Comment by blink1073** [comment]
Yeah, I looked at the classic notebook to make sure they were the same.

**Comment by blink1073** [comment]
We probably want to make a PR to `ipykernel` to update those labels and use `https` where appropriate.

**Comment by ian-r-rose** [comment]
Woah, look at all them new tests.

**Comment by blink1073** [comment]
Same tests, just spread out like peanut butter  :)

**Comment by blink1073** [comment]
Good to go, thanks!

**Pull request merged**